custom_components/eventsensor/common.py

```diff
@@ -111,22 +113,11 @@ def extract_state_from_event(state_key: str, event_data: dict):
     """
     Extract information from the event data to make a new sensor state.
 
-    Use 'dot syntax' to point for nested attributes, like `service_data.entity_id`
+    Use pydash string path for nested data.
+    https://pydash.readthedocs.io/en/latest/api.html#pydash.objects.get
     """
-    if state_key in event_data:
-        return event_data[state_key]
-    elif state_key.split(".")[0] in event_data:
-        try:
-            nested_data = event_data
-            for level in state_key.split("."):
-                nested_data = nested_data[level]
-            # Don't use dicts as state!
-            if isinstance(nested_data, dict):
-                return str(nested_data)
-            return nested_data
-        except (IndexError, TypeError):
-            pass
-    return "bad_state"
+    result = str(get(event_data, state_key, "bad_state"))
```

Comment:
It is a very nice tool, and also very light :)

Not a fan of adding requirements which are not already included in HA, but it could be useful and make code here much simpler, so ok with it.

But, just a little problem there:
* If **returning always a string**, then there is no possibility of a numeric sensor :(, and the part of the state mapping **won't work** without changes (there is a numeric parsing around there).
* Without the str casting, and respecting the old comment (`# Don't use dicts as state!`), which is important to not break a lot of things inside the HA State Machine :), everything should work all right

Well, after seeing the event data new example with an array inside, please change to `if isinstance(nested_data, (dict, list)): return str(nested_data)`, because the same applies to lists.


And even something more :)

If would be a shame to use the tool just here, and not in the ugly and recursive method `check_dict_is_contained_in_another` from `.common.py`, don't you think? :)